content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,229 @@
+low-level, standards-compliant CNI implementation present in Kubernetes clusters to offer the high-level policy and security
+controls users need at scale.
+
+Basically, before you can do things like secure pod traffic with mTLS and apply lots of high-level authentication and authorization policy
+at the service mesh layer, you must have a functional Kubernetes cluster with a functional CNI implementation, to make sure the
+basic networking pathways are set up so that packets can get from one pod to another (and from one node to another) in
+your cluster.
+
+Though some service meshes may also ship and require their own in-house CNI implementation, the Istio project has chosen not
+to, for maximum compatibility with managed offerings, cross-vendor support, and composability with the broader CNCF ecosystem.
+
+### Traffic redirection in ambient alpha
+
+The [Istio-cni](/docs/setup/additional-setup/cni/) component is an optional component in the sidecar data plane mode,
+commonly used to remove the [requirement for the `NET_ADMIN` and `NET_RAW` capabilities](/docs/ops/deployment/requirements/) for
+users deploying pods into the Istio mesh. With the launch of ambient, istio-cni has become a required component for ambient
+data plane mode. Whenever pods are added to ambient mesh, the istio-cni component configures traffic redirection for all
+incoming and outgoing traffic between the application pods and their co-located [ztunnel](/blog/2023/rust-based-ztunnel/) on
+the node via the node-level network namespace. The key difference between sidecar and ambient on the traffic redirection is
+the redirection in ambient is across network namespaces, from the application namespace to the Istio’s system namespace
+where ztunnel typically runs.
+
+As we tested more broadly in multiple Kubernetes environments which many of them have its default CNI, it became clear that
+the host network based traffic redirection mechanism we had been using for Istio ambient traffic capture and redirection
+during the alpha stage was not going to meet our requirements. The following challenges were identified:
+- The default CNI’s node-level networking configuration could interfere with the node-level networking configuration
+from istio-cni, whether it is with eBPF or iptables.
+- If users deploy a network policy for the default CNI, the network policy may not be enforced when istio-cni is deployed.
+
+### Thought process on solving the challenges
+
+At Solo, we have been brainstorming a lot on these top challenges in Ambient. We realized that applying any traffic
+routing/networking rules in the node-level network namespace invites unresolvable conflicts/incompatibilities with
+CNIs (which might use a wide variety of fundamentally networking topologies on the node) - they *must* configure traffic
+routing/networking rules in the node-level network namespace. We realized any eBPF implementation would have the same
```

Comment:
s/fundamentally/fundamental/

Isn't this just what you said in the point above?  
I take from this "we could end up in a race condition with the lower level CNI".  Is that right?